projects/dsp-ui/src/lib/action/pipes/string-transformation/truncate.pipe.ts

```diff
@@ -0,0 +1,39 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+/**
+ * This pipe can be used to shorten long text by a defined length.
+ *
+ * In markup:
+ *
+ * {{ str | dspTruncate:['24'] }}
+ *
+ * or
+ *
+ * {{ str | dspTruncate:['24', '...'] }}
+ *
+ *
+ * The first parameter defines the length where to truncate the string.
+ * Second optional parameter defines the characters to append to the shortened string. Default is '...'.
+ *
+ * The advantage of this pipe over the default Angular slice pipe is the simplicity of adding
+ *  additional characters at the end of the shortened string.
+ * The same construct with Angular slice pipe looks as follow: `{{ (str.length>24)? (str | slice:0:24)+'...':(str) }}`.
+ *
+ */
+@Pipe({
+    name: 'dspTruncate'
+})
+export class TruncatePipe implements PipeTransform {
+
+    transform(value: string, args: string[]): string {
+
+        if (!value || value.length === 0) {
```

Comment:
I think you could also explicitly check like this `typeof value !== "string" || value.length === 0`